### DISCUSSION TITLE
Using WebAssembly (wasm) to create a web app based on source code from a wails app

### DISCUSSION THREAD

**Discussion by tataDan** [wailsapp/wails#3309] [Ideas]
I started thinking about how a web app could be created reusing the go code from a wails app .  Using [this tutorial](https://golangbot.com/webassembly-using-go/) and [it's subsequent tutorial](https://golangbot.com/go-webassembly-dom-access/) as a guide, I created a web app based on the [Hello World tutorial](https://wails.io/docs/tutorials/helloworld) app.  The repository for the source code is [here](https://github.com/tataDan/wails-to-wasm/).  The repository for the app is [here](https://github.com/tataDan/wails-to-wasm-app/). The web page is [here](https://tatadan.github.io/wails-to-wasm-app/).

At this point in time I am not clear whether or not this is a feasible approach.  If not, what would be a more feasible approach?

**Comment by marcio199226** [comment]
Yeah you can do it without major issues I'm porting my mchat application to share a golang pkg responsible for encryption to wasm and then deploy my "app wails frontend" as normal spa/pwa.

Your approach is good but in my opinion you shouldn't operate on dom directly from wasm code it's better to communicate with fe as you do it with golang backend and then operate on dom using your fe framework or even better you can use the same method's signratures and return the same outputs from methods so you have only to change the import part and the rest of the code works.

Another thing to take into account that golang wasm are large due to gc and other stuff which golang brings to wasm, there is tinygo but be sure that packages you are using are supported by tinygo.
